ros2topic/ros2topic/verb/list.py

```diff
@@ -34,19 +46,31 @@ def add_arguments(self, parser, cli_name):
         parser.add_argument(
             '--include-hidden-topics', action='store_true',
             help='Consider hidden topics as well')
+        parser.add_argument(
+            '-v', '--verbose', action='store_true',
+            help='list full details about each topic')
 
     def main(self, *, args):
+        topic_info = []
         with NodeStrategy(args) as node:
             topic_names_and_types = get_topic_names_and_types(
                 node=node,
                 include_hidden_topics=args.include_hidden_topics)
+            for (topic_name, topic_types) in topic_names_and_types:
+                pub_cnt = node.count_publishers(topic_name)
+                sub_cnt = node.count_subscribers(topic_name)
+                topic_info.append((topic_name, topic_types, pub_cnt, sub_cnt))
```

Comment:
To me, having a list of tuples with four elements isn't desirable. In `topic_names_and_types`, there are two and they're right there in the variable name. Maybe there is a better way to do this. Perhaps, just compute them as you need them in the `show_topic_info` function.

Reply:
Why I compute them here is that I need the NodeStrategy. I'm not sure how to pass NodeStrategy to show_topic_info is a better way.
```py
def show_topic_info(args, topic_info, isPub):
    with NodeStrategy(args) as node:
....
```

Reply:
@audrow  I add the if-condition to avoid running count_publishers and count_subscribers while there is no verbose. I'm not sure whether it's a better way, but I try to avoid pass args into another function just for `with NodeStrategy(args) as node`.
What do you think about it?

Reply:
Sounds good.